rust/operator-binary/src/controller.rs

```diff
@@ -1511,14 +1515,25 @@ fn create_tls_volume(
     volume_name: &str,
     tls_secret_class: &str,
     requested_secret_lifetime: &Duration,
+    service_scopes: Option<Vec<String>>,
 ) -> Result<Volume> {
+    let mut secret_volume_source_builder = SecretOperatorVolumeSourceBuilder::new(tls_secret_class);
+
+    secret_volume_source_builder
+        .with_pod_scope()
+        .with_node_scope()
+        .with_format(SecretFormat::TlsPkcs12)
+        .with_auto_tls_cert_lifetime(*requested_secret_lifetime);
+
+    if let Some(scopes) = &service_scopes {
+        for scope in scopes {
+            secret_volume_source_builder.with_service_scope(scope);
+        }
+    }
```

Comment:
Can we double check this?

As far as I understood, we should have:

```
.with_pod_scope() // for the headless service
.with_listener_volume_scope(VOLUME_NAME) // for external access
```

Reply:
done here https://github.com/stackabletech/trino-operator/pull/766/commits/f68f42394638bf9ecbbbf0fb221f86a44986c392